client/src/utils/request.js

```diff
@@ -62,3 +62,21 @@ export const reorderRequests = (
 
   return newRequests;
 };
+
+export const getFileName = (path) => {
+  const pathArray = path.split("/");
+  return pathArray[pathArray.length - 1];
+};
+
+export const validateFileName = (request, name) => {
```

Comment:
Makes sense to validate if a filename is duplicated, but throwing an error is not very helpful for user without recovery. What do you think about prevent user to press submit button on frontend if the name is duplicated?

Reply:
Edit: This was before I saw the toaster. I think it's fine as is, but I think it's better UI/UX if the app doesn't allow you to submit in the first place (disabling the submit button) if there are no filename / duplications